### PR TITLE
[🐧] lsthiros' Lua port for 1.x

### PR DESCRIPTION
**Pull request by vadosnaprimer**
This was finished long ago and we've been mentioning it all the time, but [the author](https://github.com/lsthiros) hasn't requested a merge, despite of the goal being basically achieved. So why not have it preserved?

~~we could save this as a new branch~~ done --yoshi

Note: [xbuild fails on linux](https://github.com/lsthiros/BizHawk/issues/1). Maybe our [Linux hero](https://github.com/YoshiRulz) could help sort this out?

### CODE REVIEW

**Comment by YoshiRulz** [comment]
FWIW it fails with MSBuild too ([xbuild is deprecated](https://www.mono-project.com/docs/about-mono/releases/5.0.0/#xbuild)). I copied `Version/Version.csproj` from master, then edited `BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj` to remove references to missing files, and it stopped here:
```
"/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj" (default target) (10:2) ->                                                                                
(CoreCompile target) -> 
  Consoles/Nintendo/N64/N64.IDebuggable.cs(5,44): error CS0234: The type or namespace name 'NativeApi' does not exist in the namespace 'BizHawk.Emulation.Cores.Nintendo.N64' (are you missing an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64.IMemoryDomains.cs(6,44): error CS0234: The type or namespace name 'NativeApi' does not exist in the namespace 'BizHawk.Emulation.Cores.Nintendo.N64' (are you missing an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64.IStatable.cs(6,44): error CS0234: The type or namespace name 'NativeApi' does not exist in the namespace 'BizHawk.Emulation.Cores.Nintendo.N64' (are you missing an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64.ITraceable.cs(7,44): error CS0234: The type or namespace name 'NativeApi' does not exist in the namespace 'BizHawk.Emulation.Cores.Nintendo.N64' (are you missing an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64Input.cs(2,44): error CS0234: The type or namespace name 'NativeApi' does not exist in the namespace 'BizHawk.Emulation.Cores.Nintendo.N64' (are you missing an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64.cs(6,44): error CS0234: The type or namespace name 'NativeApi' does not exist in the namespace 'BizHawk.Emulation.Cores.Nintendo.N64' (are you missing an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64Audio.cs(3,44): error CS0234: The type or namespace name 'NativeApi' does not exist in the namespace 'BizHawk.Emulation.Cores.Nintendo.N64' (are you missing an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64SyncSettings.cs(3,44): error CS0234: The type or namespace name 'NativeApi' does not exist in the namespace 'BizHawk.Emulation.Cores.Nintendo.N64' (are you missing an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64VideoProvider.cs(3,44): error CS0234: The type or namespace name 'NativeApi' does not exist in the namespace 'BizHawk.Emulation.Cores.Nintendo.N64' (are you missing an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64SyncSettings.cs(63,10): error CS0246: The type or namespace name 'VideoPluginSettings' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64Input.cs(53,43): error CS0246: The type or namespace name 'mupen64plusApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64Input.cs(8,11): error CS0246: The type or namespace name 'mupen64plusInputApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64Audio.cs(46,19): error CS0246: The type or namespace name 'mupen64plusApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64Audio.cs(12,11): error CS0246: The type or namespace name 'mupen64plusAudioApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64Audio.cs(14,11): error CS0246: The type or namespace name 'mupen64plusApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64VideoProvider.cs(19,27): error CS0246: The type or namespace name 'mupen64plusApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64VideoProvider.cs(19,48): error CS0246: The type or namespace name 'VideoPluginSettings' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64VideoProvider.cs(10,11): error CS0246: The type or namespace name 'mupen64plusVideoApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64VideoProvider.cs(11,11): error CS0246: The type or namespace name 'mupen64plusApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64.IMemoryDomains.cs(16,46): error CS0246: The type or namespace name 'mupen64plusApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64.ITraceable.cs(15,11): error CS0246: The type or namespace name 'mupen64plusApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
  Consoles/Nintendo/N64/N64.cs(154,11): error CS0246: The type or namespace name 'mupen64plusApi' could not be found (are you missing a using directive or an assembly reference?) [/home/yoshi/Downloads/bhlua/BizHawk.Emulation.Cores/BizHawk.Emulation.Cores.csproj]
```
Looks like whatever was in those files is actually important. Gonna try copying *those* from master too.

**Comment by Sappharad** [comment]
The changes from his PR actually build just fine for me, after fixing the Version.csproj thing and updating the project references to Reference the /References/ instead of a random /obj/ folder.

I can't actually build / test the Lua .dylib needed to run though, because for some reason Apple removed the x86 compilers from their dev tools last fall and there's no way to get them back without downgrading to the 2017 release. So I have no way of confirming that it actually works, but the build runs and blows up when you try to use Lua without the .dylib.

Wasn't someone working on porting a managed Lua implementation to BizHawk 2 instead? Whatever happened to that one? Do we still need this even?

In summary, I can build the C# side of this but I lack the tools to build the actual C++ Lua project right now and I'm not sure if it's worth wasting a few hours to set up a dev environment in a VM to do it.

**Comment by vadosnaprimer** [comment]
I wanted to backport managed lua to 1.x branch IIRC, but never did. That would hopefully make lua work in portable branch. In 2.x there's already the NLua option which is pure C#.
